### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by jonas-45**
In this branch, we extracted some pieces of logic into separate classes;

- [ ] creation class
- [ ] listings class
- [ ] choose class

We ensured that the main class follows the "Single-responsibility Principle" by starting the code and asking the user for the initial choice of options.

We fixed all linter errors.

### CODE REVIEW

File: creation.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+  def create_rental
+    students = Student.all
+    teachers = Teacher.all
+    persons = students + teachers
+    books = Book.all
+    if persons.empty?
+      puts 'There are no People added yet.'
+    else
+      @r_person = choose_person(persons)
```
Comment (brytebee) [inline creation.rb]: Great job refactoring your code. 👏👏

However, the code throws the bug below when I tried to create a rental. Kindly fix it.

image

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @jonas-45 & @frempongdev👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great DB setup.✔️
- App functions as it should.✔️

##

Good job so far coming so far with the progress you've made! 💪💪
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: creation.rb (hunk 1 above)
Comment (frempongdev) [reply creation.rb]: Thanks for pointing it out. It has been fixed 👌🏿

**Review by rustam-tolipov — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Developer-512px-10

## Highlights
1. No Linters errors 💯
2. All requirements are complete 💯
3. Well-implemented code 💯 
4. Great readme description 🥇 

## Optional suggestions
No optional suggestions :)

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**